sandbox/install-client-archives.sh

```diff
@@ -29,8 +29,14 @@ for PACKAGE in "${CLIENT_PACKAGES[@]}"; do
     wget ${BASE_URL}${VERSION}/${PACKAGE}-${VERSION}-all.tar.gz \
         || wget ${BASE_URL}v${VERSION}/${PACKAGE}-${VERSION}-all.tar.gz
     tar xvzf ${PACKAGE}-${VERSION}-all.tar.gz
-    pip install -r ${PACKAGE}-${VERSION}-all/requirements.txt
-    pip install -r ${PACKAGE}-${VERSION}-all/requirements_gui.txt
+    REQUIREMENTS=${PACKAGE}-${VERSION}-all/requirements.txt
+    REQUIREMENTS_GUI=${PACKAGE}-${VERSION}-all/requirements_gui.txt
+
+    # requirements vary across the Python packages
+    if [ -f ${REQUIREMENTS_GUI} ]; then
+        pip install -r ${REQUIREMENTS_GUI}
+    fi
+    pip install -r ${REQUIREMENTS}
```

Comment:
This might be overly nitpicky, but I think this line should come first, before the GUI requirements.

My concern is the GUI upstream requirements may overlap with the non-GUI ones, which could lead to us installing different versions of those upstream dependencies than are specified in the main requirements file.

Reply:
Nope, I don't think that's overly nitpicky. It's very useful feedback and makes total sense, so thanks for pointing it out! Done.